### PR TITLE
Add sleep fn

### PR DESCRIPTION
**Pull request by SomeHats**
There are a bunch of places we do something like `await new Promise(resolve => setTimeout(resolve, 1000))`. Most of the time, these places have to come with an eslint disable. We have a rule enforcing that timers come from `editor.timers`, but in those cases this won't do what you would expect: when the editor is disposed, those timers will never resolve, which isn't really what you want if you're using the timers for control-flow around more expensive resources that need to be disposed of.

For example:
```tsx
const thing = createAnExpensiveResourceThatNeedsToBeExplicitlyDestroyed()
try {
    await doSomethingWithIt(thing)
    await new Promise(r => editor.timers.setTimeout(r, 100))
} finally {
    thing.dispose()
}
```

In this example, if the editor is disposed, then the `setTimeout` callback is never called, and the promise never resolves. The code in the finally block never runs, and the memory for `thing` leaks. 

### Change type

- [x] `other`

### Release notes

(internal-only change)

### CODE REVIEW

File: apps/docs/utils/ContentVectorDatabase.ts [hunk 1]
```diff
@@ -139,7 +140,7 @@ export class ContentVectorDatabase {
 		}
 
 		// Sleep for 50ms or so to avoid rate limiting
-		await new Promise((r) => setTimeout(r, 35))
+		await sleep(35)
```
Comment (mimecuvalo) [inline apps/docs/utils/ContentVectorDatabase.ts:143]: can you update the comment here? doesn't match

File: apps/docs/components/docs/docs-feedback-widget.tsx [hunk 2]
```diff
@@ -17,9 +17,8 @@ async function submitFeedback(
 	_feedback: 1 | -1 | 0,
 	_note: string
 ) {
-	const feedback: { value: number; message?: string } = { value: _feedback }
-	if (_note !== '') feedback.message = _note
-	track('Feedback', feedback)
+	// todo
+	await sleep(2000)
```
Comment (mimecuvalo) [inline apps/docs/components/docs/docs-feedback-widget.tsx:21]: ?

**Review by mimecuvalo — APPROVED** [review]
nice one! pre-approving modulo the couple comments (the feedback-widget in particular)

File: apps/docs/components/docs/docs-feedback-widget.tsx (hunk 2 above)
Comment (SomeHats) [reply apps/docs/components/docs/docs-feedback-widget.tsx:21]: oh good catch, this was a bad merge, sorry!

**Pull request merged**